src/core/BufferAttribute.js

```diff
@@ -394,13 +394,33 @@ Object.assign( BufferAttribute.prototype, {
 
 	toJSON: function () {
 
-		return {
+		const data = {
 			itemSize: this.itemSize,
 			type: this.array.constructor.name,
 			array: Array.prototype.slice.call( this.array ),
 			normalized: this.normalized
 		};
 
+		if ( this.name !== '' ) {
+
+			data.name = this.name;
+
+		}
+
+		if ( this.usage !== StaticDrawUsage ) {
+
+			data.usage = this.usage;
+
+		}
+
+		if ( this.updateRange.offset !== 0 || this.updateRange.count !== - 1 ) {
+
+			data.updateRange = this.updateRange;
+
+		}
```

Comment:
I export `.name`, `.usage`, and `.updateRange` only if they are non-default values because

1. Probably in most use cases they are default (= save space)
2. Keep compatibility. If they are required (in `BufferGeometryLoader.parse()`) older serialized buffer attributes will no longer be able to be deserialized.

Sounds good?

Reply:
I would keep this simple similar to `Texture.toJSON()` and not add these checks.

> If they are required (in BufferGeometryLoader.parse()) older serialized buffer attributes will no longer be able to be deserialized.

I don't think this breaks code as long as `parse()` check for `undefined`.

Reply:
Hm, but some modules like `BufferGeometry` or `Material` don't export empty name (I guess because of saving space) so I think it would be ok. But not strong opinion. @mrdoob Any preference?

Reply:
No strong opinion either. I'm fine with whatever @mrdoob wants 👍 .